homeassistant/components/trend/binary_sensor.py

```diff
@@ -96,21 +103,50 @@ async def async_setup_platform(
         max_samples = device_config[CONF_MAX_SAMPLES]
         min_gradient = device_config[CONF_MIN_GRADIENT]
         sample_duration = device_config[CONF_SAMPLE_DURATION]
+        min_samples = device_config[CONF_MIN_SAMPLES]
 
-        sensors.append(
-            SensorTrend(
-                hass,
-                device_id,
-                friendly_name,
-                entity_id,
-                attribute,
-                device_class,
-                invert,
+        # check if min_samples is smaller than max_samples
+        issue = f"min_samples_larger_max_samples_{device_id}"
+        if min_samples > max_samples:
```

Comment:
Please remove all of this, and instead add a custom validator to the schema which checks this:
```py
def _validate_min_max(data: dict[str, Any]) -> dict[str, Any]:
    if not CONF_MIN_SAMPLES in data or not CONF_MAX_SAMPLES in data:
        return data
    if data[CONF_MAX_SAMPLES] < data[CONF_MIN_SAMPLES]:
        raise vol.Invalid("min_samples must be smaller than max_samples")

SENSOR_SCHEMA = vol.All(
    vol.Schema(
        ...
    ),
    _validate_min_max
)
```

That will give you a nice error message like this:
```
2023-11-21 22:55:41.483 ERROR (MainThread) [homeassistant.config] Invalid config for 'binary_sensor.trend' at configuration.yaml, line 175: min_samples must be smaller than max_samples for dictionary value 'sensors->cpu_speed', got {'entity_id': 'sensor.cpu_speed', 'min_samples': 3, 'max_samples': 2}, please check the docs at https://www.home-assistant.io/integrations/trend
```

Reply:
Oh thanks for this hint, didn't know that.